### DISCUSSION TITLE
Uploading HLS files to S3 buckets when stream is running.

### DISCUSSION THREAD

**Discussion by Mohit-3196** [ant-media/Ant-Media-Server#5338] [FAQs] [answered]
I want to upload my HLS files to the S3 bucket.
Currently, the files are stored in the bucket after the stream ends. But my use case requires storing them when the stream is running.
I checked and I saw that I can use [fuse mechanisms](https://github.com/orgs/ant-media/discussions/5090) to achieve that but it seems a little complicated.

Are there some other solutions/settings on the server side itself that I can use to achieve this?

**Comment by zerostack0** [comment, marked as answer]
Thank you for your question.

We can easily upload HLS files to the S3 bucket periodically and there is a server-side solution for this starting AMS v2.6.3.
To do this:

1. Add the S3 bucket details where you would like to upload the HLS files
![Screenshot 2023-07-10 140510](https://github.com/ant-media/Ant-Media-Server/assets/86982446/76f7780b-1ad3-485a-a603-600dd467b9a6)

2. Open [advanced settings](https://github.com/orgs/ant-media/discussions/5161#discussioncomment-6401677) on the dashboard (which came on v2.6.2) and set
`settings.hlsHttpEndpoint=https://{SERVER_DOMAIN}:5443/{APPLICATION_NAME}/hls-upload` 

3. Once the streaming to the application has started, AMS will upload the HLS files to provide S3 bucket.

Thank you.
